File: frontend/src/types/component.ts
export interface Component {
  id: number;
  name: string;
  description?: string;
  manufacturer_part_number?: string;
  supplier_part_number?: string;
  mpn?: string;
  manufacturer?: string;
  category?: string;
  subcategory?: string;
  package_type?: string;
  quantity: number;
  min_quantity: number;
  location?: string;
  datasheet_url?: string;
  unit_price?: number;
  supplier?: string;
  notes?: string;
  category_id?: number | null;
  footprint_id?: number | null;
  tags?: string | null;
  ipn?: string | null;
  is_favorite?: boolean;
  status?: string | null;
  category_name?: string | null;
  footprint_name?: string | null;
  min_order_quantity?: number | null;
  created_at: string;
  updated_at: string;
}

export interface ComponentCreate {
  name: string;
  description?: string;
  manufacturer_part_number?: string;
  supplier_part_number?: string;
  mpn?: string;
  manufacturer?: string;
  category?: string;
  subcategory?: string;
  package_type?: string;
  quantity?: number;
  min_quantity?: number;
  location?: string;
  datasheet_url?: string;
  unit_price?: number;
  supplier?: string;
  notes?: string;
  category_id?: number | null;
  footprint_id?: number | null;
  tags?: string;
  ipn?: string;
  is_favorite?: boolean;
  status?: string;
  min_order_quantity?: number;
}

export type ComponentUpdate = Partial<ComponentCreate>;

export interface StockTransaction {
  id: number;
  component_id: number;
  quantity_change: number;
  reason: string;
  reference?: string;
  created_at: string;
}

export const COMPONENT_CATEGORIES = [
  "Resistor",
  "Capacitor",
  "Inductor",
  "Diode",
  "Transistor",
  "IC",
  "Connector",
  "LED",
  "Crystal",
  "Relay",
  "Switch",
  "Fuse",
  "Sensor",
  "Module",
  "PCB",
  "Mechanical",
  "Cable",
  "Other",
] as const;

File: backend/app/schemas/component.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class ComponentBase(BaseModel):
    name: str
    manufacturer_part_number: Optional[str] = None
    supplier_part_number: Optional[str] = None
    description: Optional[str] = None
    category: Optional[str] = None
    subcategory: Optional[str] = None
    quantity: int = 0
    min_quantity: int = 0
    location: Optional[str] = None
    datasheet_url: Optional[str] = None
    unit_price: Optional[float] = None
    supplier: Optional[str] = None
    supplier_url: Optional[str] = None
    package_type: Optional[str] = None
    notes: Optional[str] = None
    # Part-DB style fields
    category_id: Optional[int] = None
    footprint_id: Optional[int] = None
    tags: Optional[str] = None
    manufacturer: Optional[str] = None
    mpn: Optional[str] = None
    ipn: Optional[str] = None
    is_favorite: bool = False
    status: str = "active"
    min_order_quantity: Optional[int] = None


class ComponentCreate(ComponentBase):
    pass


class ComponentUpdate(BaseModel):
    name: Optional[str] = None
    manufacturer_part_number: Optional[str] = None
    supplier_part_number: Optional[str] = None
    description: Optional[str] = None
    category: Optional[str] = None
    subcategory: Optional[str] = None
    quantity: Optional[int] = None
    min_quantity: Optional[int] = None
    location: Optional[str] = None
    datasheet_url: Optional[str] = None
    unit_price: Optional[float] = None
    supplier: Optional[str] = None
    supplier_url: Optional[str] = None
    package_type: Optional[str] = None
    notes: Optional[str] = None
    # Part-DB style fields
    category_id: Optional[int] = None
    footprint_id: Optional[int] = None
    tags: Optional[str] = None
    manufacturer: Optional[str] = None
    mpn: Optional[str] = None
    ipn: Optional[str] = None
    is_favorite: Optional[bool] = None
    status: Optional[str] = None
    min_order_quantity: Optional[int] = None


class ComponentResponse(ComponentBase):
    id: int
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None
    category_name: Optional[str] = None
    footprint_name: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)


class StockTransactionResponse(BaseModel):
    id: int
    component_id: int
    quantity_change: int
    reason: str
    reference_id: Optional[int] = None
    created_at: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)


class ComponentListResponse(BaseModel):
    items: list[ComponentResponse]
    total: int
    page: int
    page_size: int
    total_pages: int
